pkg/kubectl/cmd/delete.go

```diff
@@ -223,7 +231,7 @@ func (o *DeleteOptions) Validate(cmd *cobra.Command) error {
 			// To preserve backwards compatibility, but prevent accidental data loss, we convert --grace-period=0
 			// into --grace-period=1 and wait until the object is successfully deleted. Users may provide --force
 			// to bypass this wait.
-			o.WaitForDeletion = true
+			o.WaitForGracefulDeletion = true
```

Comment:
The problem is that I cannot use WaitForDeletion at line 289 in reapResultWithWaitingOptions method.
If I do, then kubectl will start sending GET requests after deleting the resource if user sets `--wait=true`.

I can remove that whole code from line 289 to 292, if all reapers wait for the resource to be deleted. PodReaper does not: https://github.com/kubernetes/kubernetes/blob/9f92832fadebc11cd057fbde2bfc00fe46f860a2/pkg/kubectl/stop.go#L517. I can update PodReaper, but I think that is a big enough change to do in a separate PR.